deploy-service/common/src/main/java/com/pinterest/deployservice/rodimus/RodimusManagerImpl.java

```diff
@@ -73,40 +72,24 @@ public boolean shouldSkipClass(Class<?> c) {
         }
     }
 
-    private boolean refreshCachedKey() throws Exception {
-        String prevKnoxKey = this.cachedKey;
-
-        if (this.fsKnox != null) {
-            this.cachedKey = new String(this.fsKnox.getPrimaryKey());
-        }
-
-        if ( prevKnoxKey == null )
-        {
-            return this.cachedKey != null;
-        }else{
-            return ! prevKnoxKey.equals( this.cachedKey );
-        }
-    }
-
     private void setAuthorization() throws Exception {
-        if (this.cachedKey == null) {
-            this.refreshCachedKey();
```

Comment:
seems like we remove the Cache handling here, is that mean that we no longer use cache in the new code

Reply:
yes, we remove the cache on Teletraan side. We relay on the Knox interface to handling key rotation and caching, as recommended by Knox documentation.